projects/storefrontapp-e2e-cypress/cypress/helpers/checkout-flow.ts

```diff
@@ -173,9 +174,7 @@ export function goToProductDetailsPage() {
 
 export function addProductToCart() {
   cy.get('cx-item-counter').findByText('+').click();
-  cy.get('cx-add-to-cart')
-    .findByText(/Add To Cart/i)
-    .click();
+  addToCart();
```

Comment:
reason why I added this function that waits for the cart to be 'refreshed' is because cypress did not wait and when we proceeded to checkout, it returned us to homepage due to the `CartNotEmptyGuard` as the cart was empty when it clicked proceed to checkout

Reply:
this applies to all instances in the application where we do 

```
  cy.get('cx-add-to-cart')
    .findByText(/Add To Cart/i)
    .click();
```